src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocMethodCheck.java

```diff
@@ -357,27 +358,13 @@ protected final void processAST(DetailAST ast) {
     }
 
     /**
-     * Some javadoc.
-     * @param methodDef Some javadoc.
-     * @return Some javadoc.
+     * Checks if the method is marked with an annotation which allows it
+     * to omit javadoc documentation.
+     * @param methodDef The method definition.
+     * @return true if the method is exempt from comments, false otherwise.
      */
     private boolean hasAllowedAnnotations(DetailAST methodDef) {
-        boolean result = false;
-        final DetailAST modifiersNode = methodDef.findFirstToken(TokenTypes.MODIFIERS);
-        DetailAST annotationNode = modifiersNode.findFirstToken(TokenTypes.ANNOTATION);
-        while (annotationNode != null && annotationNode.getType() == TokenTypes.ANNOTATION) {
-            DetailAST identNode = annotationNode.findFirstToken(TokenTypes.IDENT);
-            if (identNode == null) {
-                identNode = annotationNode.findFirstToken(TokenTypes.DOT)
-                    .findFirstToken(TokenTypes.IDENT);
-            }
-            if (allowedAnnotations.contains(identNode.getText())) {
-                result = true;
-                break;
-            }
-            annotationNode = annotationNode.getNextSibling();
-        }
-        return result;
+        return AllowedAnnotationsUtil.hasAllowedAnnotations(methodDef, allowedAnnotations);
```

Comment:
please inline this method to place where hasAllowedAnnotations is used, no need to wrappers, that keep same name.

Reply:
Please notice that inlining means I will be repeating the "allowedAnnotations" second argument on every inlined call. To avoid repetition, I think it's worth having this extra method. If you insist I will change it.

Reply:
fixed!